core/src/main/scala/chisel3/Mem.scala

```diff
@@ -246,6 +246,10 @@ sealed abstract class MemBase[T <: Data](val t: T, val length: BigInt)
   )(
     implicit compileOptions: CompileOptions
   ): T = {
+    require(
+      Builder.currentModule == _parent,
+      s"cannot create a memory port in a different module (${Builder.currentModule.get.name}) than where the memory is (${_parent.get.name})."
+    )
```

Comment:
```suggestion
    if (Builder.currentModule != _parent) {
      throwException(s"cannot create a memory port in a different module (${Builder.currentModule.get.name}) than where the memory is (${_parent.get.name}).")
    }
```
Stylistically, `require` should be reserved for incorrect arguments given to a function by a user. Neither of the checked things are arguments so this is more of just a normal invalid state.